docs/reference/event-dependent-configuration.md

```diff
@@ -40,9 +51,146 @@ For example, this event has five top-level fields (agent, ip, request, response,
 }
 ```
 
-To reference the `os` field, specify `[ua][os]`. To reference a top-level field such as `request`, you can simply specify the field name.
+To reference the `os` field, specify `[ua][os]`. 
+To reference a top-level field such as `request`, specify the field name.
+
+
+### Why use field references? [field-reference-deep-dive]
+
+You might find situations in which you need to refer to a field or collection of fields by name. 
+You can accomplish this goal using the Logstash field reference syntax.
+
+The syntax to access a field specifies the entire path to the field, with each fragment wrapped in square brackets.
+When a field name contains square brackets, the brackets must be properly [ escaped](#formal-grammar-escape-sequences).
+
+Field references can be expressed literally within [conditional statements](#conditionals) in your pipeline configurations,
+as string arguments to your pipeline plugins, or within sprintf statements that will be used by your pipeline plugins:
+
+```
+filter {
+  #  +----literal----+     +----literal----+
+  #  |               |     |               |
+  if [@metadata][date] and [@metadata][time] {
+    mutate {
+      add_field {
+        "[@metadata][timestamp]" => "%{[@metadata][date]} %{[@metadata][time]}"
+      # |                      |    |  |               |    |               | |
+      # +----string-argument---+    |  +--field-ref----+    +--field-ref----+ |
+      #                             +-------- sprintf format string ----------+
+      }
+    }
+  }
+}
+```
+
+
+### Formal grammar [formal-grammar]
+
+Expand the section below if you want more details about formal grammar for field references.
```

Comment:
I love the drop-down, I think it works well, and ensures that the standard user flow is uninterrupted

I'm not quite sure what the right answer is here, but I'm not sure if we need this line, or if it could be re-worded a little to reduce the repetition of "Formal Grammar for field references."


```suggestion
Expand the section below if you would like to explore the formal grammar.
```